src/assets/TypeScriptAsset.js

```diff
@@ -28,9 +31,41 @@ class TypeScriptAsset extends Asset {
 
     // Overwrite default if config is found
     if (tsconfig) {
+      // Files read by TypeScript parser while resolving "extends"
+      let readFiles = [];
+      // https://github.com/Microsoft/TypeScript/blob/62306bc3f9e119dc17b400ff263dad532da6cdb1/scripts/build/upToDate.js#L370
+      const parseConfigHost = {
+        useCaseSensitiveFileNames: true,
+        getCurrentDirectory: () => process.cwd(),
+        readDirectory: file => fs.readdirSync(file),
+        fileExists: file => fs.existsSync(file) && fs.statSync(file).isFile(),
+        readFile: file => {
+          readFiles.push(file);
+          return fs.readFileSync(file, 'utf8');
+        },
+        onUnRecoverableConfigFileDiagnostic: () => undefined
+      };
+      // Get directory of tsconfig.json
+      const tsconfigDir = path.dirname(
+        await config.resolve(this.name, ['tsconfig.json'])
+      );
+      // Parse contents of tsconfig.json with TypeScript API to resolve "extends"
+      const parsedTsconfig = typescript.parseJsonConfigFileContent(
+        tsconfig,
+        parseConfigHost,
+        tsconfigDir
+      );
+      // Add resolved files to watch list
+      for (const file of readFiles) {
+        if (file.endsWith('tsconfig.json')) {
```

Comment:
What other files can be read here? Should they be added as dependencies as well?

Reply:
Reading TypeScript source, it only reads `tsconfig` files. I just added the assertion in case TypeScript does breaking changes regarding to this, so it prevents unexpected behaviour.